src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -489,8 +490,13 @@ public void updateParticipant(Study study, StudyParticipant participant) {
                 externalIdService, substudyService, study, false);
         Validate.entityThrowingException(validator, participant);
         
-        Set<Roles> callerRoles = BridgeUtils.getRequestContext().getCallerRoles();
+        // FindBugs: Null passed for non-null parameter of BridgeUtils.collectExternalIds(Account).
+        // In actuality this will not happen (an exception will be thrown first).
+        if (account == null) {
+            return;
+        }
```

Comment:
Similarly, if we know for sure that this will never be null, would it be better to exclude the warning than to write unreachable code?

Reply:
Okay this one forced me to rethink what I was doing. Apparently all these shenanigans are to throw a 404 error rather than an invalid entity exception, but the validator catches the invalid substudy of the caller correctly, and this can be simplified if you just leave it with the invalid entity exception. And I don't see now why I insisted on a 404. So: cleaned up code, findbug errors go away, tests adjusted.